### PR TITLE
Fix bug on study creation error

### PR DESCRIPTION
**Pull request by etiennehomer**

### CODE REVIEW

File: src/components/create-study-form.js [hunk 1]
```diff
@@ -273,12 +272,13 @@ export const CreateStudyForm = (props) => {
...
             activeDirectory
         ).then((res) => {
-            setCreateStudyErr('');
-            setStudyName('');
-            setStudyDescription('');
-            dispatch(removeSelectedFile());
-
-            if (!res.ok) {
+            if (res.ok) {
+                props.setOpen(false);
```
Comment (sBouzols) [inline src/components/create-study-form.js]: strange thing to edit props inside the component isn't it ?
If the 'open' state is outside this component, then we should add a `onClose` props to pass to this component and then call it nope ?

Comment (etiennehomer) [reply src/components/create-study-form.js]: done

File: src/components/create-contingency-list-form.js [hunk 2]
```diff
@@ -75,7 +81,7 @@ export const CreateContingencyListForm = (props) => {
 
         let isPrivateContingencyList = contingencyListPrivacy === 'private';
 
-        props.setOpen(false);
+        onClose();
```
Comment (sBouzols) [inline src/components/create-contingency-list-form.js]: For homogenous code, you should move this call in a `if(res.ok)` in the Promise.then bellow.
Same thing that you did in create-study-form

**Review by sBouzols — CHANGES_REQUESTED** [review]

File: src/components/tree-views-container.js [hunk 3]
```diff
@@ -83,7 +81,7 @@ const TreeViewsContainer = () => {
     const activeDirectory = useSelector((state) => state.activeDirectory);
     const userId = useSelector((state) => state.user.profile.sub);
 
-    const currentEdit = useRef(null);
+    // const currentEdit = useRef(null);
```
Comment (sBouzols) [inline src/components/tree-views-container.js]: remove this

File: src/components/create-filter-form.js [hunk 4]
```diff
@@ -86,12 +89,13 @@ const CreateFilterDialog = ({
     title,
     customTextValidationBtn,
     customTextCancelBtn,
-    action,
+    activeDirectory,
```
Comment (sBouzols) [inline src/components/create-filter-form.js]: this activeDirectory is in the redux/store, you could avoid passing it in props
`const activeDirectory = useSelector((state) => state.activeDirectory);`

File: src/components/create-filter-form.js [hunk 5]
```diff
@@ -107,18 +111,38 @@ const CreateFilterDialog = ({
...
+                transient: true,
+            },
+            newNameList,
+            newListType,
+            filterPrivacy === 'private',
+            activeDirectory
+        ).then((res) => {
+            if (res.ok) {
+                onClose();
+                setCreateFilterErr('');
```
Comment (sBouzols) [inline src/components/create-filter-form.js]: reset `setNewListName('')` and setNewListType('SCRIPT') as well maybe ?

**Review by sBouzols — CHANGES_REQUESTED** [review]

File: src/components/tree-views-container.js (hunk 3 above)
Comment (etiennehomer) [reply src/components/tree-views-container.js]: done

File: src/components/create-filter-form.js (hunk 4 above)
Comment (etiennehomer) [reply src/components/create-filter-form.js]: done

File: src/components/create-filter-form.js (hunk 5 above)
Comment (etiennehomer) [reply src/components/create-filter-form.js]: i created a resetDialog for filter, contingencies and studies that reset all fields with the default values

**Review by sBouzols — APPROVED** [review]
fine code review
No tests

File: src/components/create-study-form.js [hunk 6]
```diff
@@ -287,10 +298,14 @@ export const CreateStudyForm = (props) => {
...
                         .then((data) => {
-                            setCreateStudyErr(data.message);
+                            setCreateStudyErr(
+                                data.error + ' - ' + data.message
+                            );
                         })
                         .catch((error) => {
-                            setCreateStudyErr(error);
+                            setCreateStudyErr(
+                                error.error + ' - ' + error.message
```
Comment (jonenst) [inline src/components/create-study-form.js]: error.error is undefined

**Review by jonenst — CHANGES_REQUESTED** [review]
error.error is undefined. For example when stop the gateway, you get:
"undefined - JSON.parse: unexpected character at line 1 column 1 of the JSON data"

**Review by jonenst — APPROVED** [review]

**Pull request merged**